### PR TITLE
Discovery: implement server API

### PR DESCRIPTION
**Pull request by reinkrul**
Implements APIs specified by RFC;
- Register presentation
- Delete presentation
- Get presentations given tag (delta)

Out of scope: search (separate PR to keep changes small)

### CODE REVIEW

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
All ok, but I think the error/status return for a wrong audience is incorrect. My guess is that it'll respond with a 500.

**Comment by reinkrul** [comment]
> All ok, but I think the error/status return for a wrong audience is incorrect. My guess is that it'll respond with a 500.

Internal server error? bad request qwould be more applicable?

File: discovery/api/v1/wapper.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		Entries: presentations,
+		Tag:     string(*newTag),
+	}, nil
+}
+
+func (w *Wrapper) RegisterPresentation(_ context.Context, request RegisterPresentationRequestObject) (RegisterPresentationResponseObject, error) {
+	err := w.Server.Add(request.ServiceID, *request.Body)
+	if err != nil {
+		return nil, err
+	}
```
Comment (gerardsn) [inline discovery/api/v1/wapper.go:82]: should this return a 404 when errors.Is(err, errRetractionReferencesUnknownPresentation)? Also see comment on https://github.com/nuts-foundation/nuts-specification/pull/267

**Review by gerardsn — CHANGES_REQUESTED** [review]

File: discovery/module.go [hunk 2]
```diff
@@ -184,29 +202,28 @@ func (m *Module) validateRetraction(serviceID string, presentation vc.Verifiable
...
-	var retractJTIString string
-	if retractJTIRaw, ok := presentation.JWT().Get("retract_jti"); !ok {
-		return errors.New("retraction presentation does not contain 'retract_jti' claim")
-	} else {
-		if retractJTIString, ok = retractJTIRaw.(string); !ok {
-			return errors.New("retraction presentation 'retract_jti' claim is not a string")
-		}
+	retractJTIRaw, _ := presentation.JWT().Get("retract_jti")
+	retractJTI, ok := retractJTIRaw.(string)
+	if !ok {
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply discovery/module.go:211]: Should return `errInvalidRetractionJTIClaim`, added check and test

File: discovery/api/v1/wapper.go (hunk 1 above)
Comment (reinkrul) [reply discovery/api/v1/wapper.go:82]: Nobody noticed this file is named `wapper.go`?

Comment (reinkrul) [reply discovery/api/v1/wapper.go:82]: Specified it as HTTP 400 with JSON error response

**Review by gerardsn — APPROVED** [review]

**Pull request merged**